### PR TITLE
roachtest: gather cores if requested

### PR DESCRIPTION
**Pull request by cucaroach**
- roachtest: allow tests to opt in to core gathering and compress cores
- roachtest: gather cores on multiregion tests that are timing out

Relates to #90683 CRDB-20887

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/91632#-:-NGTWu8nDYFED5NRZxtI:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @cucaroach, @herkolategan, @rytaft, @smg260, and @srosenberg)

___
*[`pkg/roachprod/vm/gce/utils.go` line 171 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/91632#-NGTWfXJ9TQWaGlvD-9X:-NGTWfXK1y3jx-CwyjJ3:b-f99m8i) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6ae9653b67071cca09ed296711cecc821fc284e/pkg/roachprod/vm/gce/utils.go#L171)):*
> ```Go
> echo "kernel.core_pattern=$CORE_PATTERN" >> /etc/sysctl.conf
> 
> sysctl --system  # reload sysctl settings
> ```

Seems like this line is removed, was this intentional?

**Review by cucaroach — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/91632)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @herkolategan, @rytaft, @smg260, @srosenberg, and @yuzefovich)

___
*[`pkg/roachprod/vm/gce/utils.go` line 171 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/91632#-NGTWfXJ9TQWaGlvD-9X:-NGTXpGcBc4BhpE1VvF_:bifzx5k) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6ae9653b67071cca09ed296711cecc821fc284e/pkg/roachprod/vm/gce/utils.go#L171)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

Seems like this line is removed, was this intentional?
</blockquote></details>

No nice catch, it isn't strictly necessary for my purposes but who knows what else it was doing so I put it back.

File: pkg/roachprod/vm/gce/utils.go [hunk 1]
```diff
@@ -211,6 +193,31 @@ for service in apport.service atd.service; do
...
+* hard core unlimited
+root soft core unlimited
+root hard core unlimited
+EOF
+
+mkdir -p /mnt/data1/cores
+chmod a+w /mnt/data1/cores
+cat <<'EOF' > /bin/gzip_core.sh
+#!/bin/sh
+exec /bin/gzip -f - > /mnt/data1/cores/core.$1.$2.$3.$4.gz
```
Comment (srosenberg) [inline pkg/roachprod/vm/gce/utils.go:209]: `pigz` might save a few mins if the core is very large. (May need to `apt-get install pigz`.)

**Review by srosenberg — APPROVED** [review]

**Review by srosenberg — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/91632)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @cucaroach, @herkolategan, @rytaft, @smg260, and @yuzefovich)

___
*[`pkg/roachprod/vm/gce/utils.go` line 171 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/91632#-NGTWfXJ9TQWaGlvD-9X:-NGTbfCF8F3VnAvPd3Qj:bml79lu) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6ae9653b67071cca09ed296711cecc821fc284e/pkg/roachprod/vm/gce/utils.go#L171)):*
> Everytime you make changes to your sysctl configuration file, you must reload it to activate the changes.

:man_shrugging:

File: pkg/roachprod/vm/gce/utils.go (hunk 1 above)
Comment (cucaroach) [reply pkg/roachprod/vm/gce/utils.go:209]: For a 300MB core its like 3s vs .3s so wow, that's like a 10x speedup but hardly seems worth it.

**Review by cucaroach — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/91632)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @herkolategan, @rytaft, @smg260, @srosenberg, and @yuzefovich)

___
*[`pkg/roachprod/vm/gce/utils.go` line 171 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/91632#-NGTWfXJ9TQWaGlvD-9X:-NGTrlmaCz0Q7G08-xW1:b-1rerww) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6ae9653b67071cca09ed296711cecc821fc284e/pkg/roachprod/vm/gce/utils.go#L171)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

> Everytime you make changes to your sysctl configuration file, you must reload it to activate the changes.

:man_shrugging: 
</blockquote></details>

Right but we already editted the /proc/sys/kernel/core_pattern file so we're good, and editting the sysctl.conf makes it stay that way upon reboot.   What does the sysctl --system add?   At a minimum it ensures the current state and the state after a reboot will be the same, so thats something.

**Pull request merged**